src/app/sidenav/sidenav.component.ts

```diff
@@ -20,49 +18,69 @@ export class MzSidenavComponent implements AfterViewInit, OnDestroy {
   @Input() edge: string;
   @Input() fixed: boolean;
   @Input() id: string;
-  @Input() width: number;
   @Input() onClose: Function;
   @Input() onOpen: Function;
+  @Input() width: number;
 
-  constructor(private renderer: Renderer) { }
+  private _opened = false;
+  private collapseButton: JQuery<Element>;
+
+  get opened() { return this._opened };
+  set opened(value: boolean) {
+    this._opened = value;
+    this.collapseButton.sideNav(this._opened ? 'show' : 'hide');
+  }
 
   ngAfterViewInit() {
     this.initCollapseButton();
     this.initCollapsibleLinks();
   }
 
   ngOnDestroy() {
-    $(`#${this.collapseButtonId}`).sideNav('destroy');
+    this.collapseButton.sideNav('destroy');
   }
 
   initCollapseButton() {
-    if (!this.collapseButtonId) {
-      return;
-    }
-
-    const collapseButton = $(`#${this.collapseButtonId}`)[0];
+    // fake button if no collapseButtonId is provided
+    this.collapseButton = this.collapseButtonId
+      ? $(`#${this.collapseButtonId}`)
+      : $(document.createElement('template'));
 
-    // Make collapse button visible on all resolution if side navigation is not fixed
+    // make collapse button visible on all resolution if side navigation is not fixed
     if (!this.fixed) {
-      this.renderer.setElementClass(collapseButton, 'show-on-large', true);
+      this.collapseButton.addClass('show-on-large');
     }
 
-    // Add data-activates to collapse button
-    this.renderer.setElementAttribute(collapseButton, 'data-activates', this.id);
+    // add data-activates to collapse button
+    this.collapseButton.attr('data-activates', this.id);
+
+    // extend onOpen function to update opened state
+    const onOpen = this.onOpen || (() => {});
+    this.onOpen = () => {
+      onOpen();
+      this._opened = true;
+    };
+
+    // extend onClose function to update opened state
+    const onClose = this.onClose || (() => {});
+    this.onClose = () => {
+      onClose();
+      this._opened = false;
+    };
 
-    // Initialize collapsible button for side navigation
-    $(collapseButton).sideNav(<any>{
+    // initialize sidenav
+    this.collapseButton.sideNav(<any>{
```

Comment:
It should use SideNavOptions interface for the type but we have to add the new options onClose and onOpen in the typing.

Reply:
I'll do another PR .. this has nothing to do with my changes